ipaserver/install/dogtaginstance.py

```diff
@@ -275,6 +279,66 @@ def uninstall(self):
             logger.critical("failed to uninstall %s instance %s",
                             self.subsystem, e)
 
+    def secure_ajp_connector(self):
+        """ Update AJP connector to use a password protection  """
+        server_xml = open(paths.PKI_TOMCAT_SERVER_XML, 'r').read()
+
+        # AJP connector is set on port 8009. Use non-greedy search to find it
+        connector = re.search('<Connector port="8009".+?/>', server_xml)
+
+        # no AJP connector means no need to update anything
+        if connector is None:
+            return
+
+        # Detect tomcat version and choose the right option name
+        # pre-9.0.31.0 uses 'requiredSecret'
+        # 9.0.31.0 or later uses 'secret'
+        secretattr = ' requiredSecret="'
+        try:
+            result = ipautil.run([paths.BIN_TOMCAT, "version"],
+                                 capture_output=True)
+            sn = re.search(
+                r'Server number:\s+([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)$',
+                result.output, flags=re.MULTILINE)
+            if sn is None:
+                logger.info("tomcat version cannot be parsed, "
+                            "default to pre-9.0.31.0")
+            else:
+                v = tasks.parse_ipa_version(sn.group(1))
```

Comment:
Could you please move the version parser to an internal helper function like ``_get_tomcat_version``?
Also ``re.MULTILINE`` is not required here.